src/library/database.c

```diff
@@ -116,8 +116,15 @@ int preconstruct_fifo(const conf_t *config)
 	int rc;
 	char err_buff[BUFFER_SIZE];
 
-	/* Make sure that there is no such file/fifo */
-	unlink_fifo();
+	/* Ensure that the RUN_DIR exists */
+	if (mkdir(RUN_DIR, 0755) && errno != EEXIST) {
```

Comment:
Addressed!

I don't think we need to zero `errno` here - we're only checking it when `mkdir()` fails, so it will be set in that case; we're really just guarding against that error being 'this directory already exists', which isn't a problem for us.

If there's a strong preference I can add `errno = 0`  though, it's not going to hurt!

Reply:
You are right with errno. No need to change.